### ISSUE TITLE
Detect import name shadowing

### ISSUE THREAD

**Issue by chavacava** [mgechev/revive#112]
In GO it is possible to declare identifiers (packages, structs, interfaces, params, receivers, variables, constants...) that conflict with the name of an imported package.

for example (from Kubernetes):
```go
package app

import (
	...
	"k8s.io/kubernetes/pkg/controller/certificates/signer"
	...
)

func startCSRSigningController(ctx ControllerContext) (http.Handler, bool, error) {
	...
	signer, err := signer.NewCSRSigningController(...)
	...
	go signer.Run(1, ctx.Stop)

	return nil, true, nil
}
```
The variable `signer` shadows the import name `signer`. This kind of shadowing makes the code harder to understand.

I propose to add a rule to spot identifiers that shadow an import.

**Comment by mgechev** [comment]
I've hit this problem quite often. This makes sense to me. Would you want to take it or we can tag it with `help wanted`?

**Comment by chavacava** [comment]
I will take it

**Comment by chavacava** [comment]
closed by PR #114
